File: allLongestStrings/all_longest_strings.py

from typing import List


def all_longest_strings(input_array: List[str]) -> List[str]:
    longest = max(map(len, input_array))
    return [n for n in input_array if len(n) == longest]


inputs = [
    ['aba', 'aa', 'ad', 'vcd', 'aba'],
    ['vvx', 'ikl', 'kpp', 'md', 'kjd'],
    ['asdfad', 'sskl', 'irojgoe', 'irjeioj', 'gojkfd'],
    ['r', 'qepotwe', 'rejwgo', 'd', 'optr'],
    ['plsdfk', 'njbjowiq', 'mweobupo', 'jeoiwzxf', 'dv']
]

for inp in inputs:
    print(f'{inp} -> {all_longest_strings(inp)}')

File: allLongestStrings/allLongestStrings.ts

// Fun with ternaries :D
const allLongestStrings = (inputArray: string[]): string[] => (
	inputArray.reduce( (total, curr) => (
		total[0] === undefined 
			? [ curr ]
			: ( total[0].length === curr.length 
					? [ ...total, curr ]
				: total[0].length > curr.length 
					? total
				: [ curr ] // current is longest 
			) 
	), [])
);

const inputs: string[][] = [
	['aba', 'aa', 'ad', 'vcd', 'aba'],
	['vvx', 'ikl', 'kpp', 'md', 'kjd'],
	['asdfad', 'sskl', 'irojgoe', 'irjeioj', 'gojkfd'],
	['r', 'qepotwe', 'rejwgo', 'd', 'optr'],
	['plsdfk', 'njbjowiq', 'mweobupo', 'jeoiwzxf', 'dv']
];

console.time('allLongestStrings x4');

inputs.forEach( inp => { 
	console.log(`${inp} -> ${allLongestStrings(inp)}`); })

console.timeEnd('allLongestStrings x4');
